src/views/orb-view.ts

```diff
@@ -364,6 +384,73 @@ export class OrbView<N extends INodeBase, E extends IEdgeBase> implements IOrbVi
     }, 1);
   };
 
+  _handleKeyDown = (event: KeyboardEvent) => {
+    if (!this._settings.interaction.keyboard?.isEnabled) {
+      return;
+    }
+
+    switch (event.key) {
+      case '-': {
+        const zoomOutFactor = this._settings.interaction.keyboard.zoomOutFactor;
+        this._zoomOut(zoomOutFactor);
+        break;
+      }
+      case '+': {
+        const zoomInFactor = this._settings.interaction.keyboard.zoomInFactor;
+        this._zoomIn(zoomInFactor);
+        break;
+      }
+      case 'ArrowLeft': {
+        const panFactor = this._settings.interaction.keyboard.panFactor;
+        const dragLeftTransform = this._renderer.getPanTransform(PanDirectionType.LEFT, panFactor);
+        this._applyTransformation(dragLeftTransform);
+        break;
+      }
+      case 'ArrowRight': {
+        const panFactor = this._settings.interaction.keyboard.panFactor;
+        const dragRightTransform = this._renderer.getPanTransform(PanDirectionType.RIGHT, panFactor);
+        this._applyTransformation(dragRightTransform);
+        break;
+      }
+      case 'ArrowUp': {
+        const panFactor = this._settings.interaction.keyboard.panFactor;
+        const dragUpTransform = this._renderer.getPanTransform(PanDirectionType.UP, panFactor);
+        this._applyTransformation(dragUpTransform);
+        break;
+      }
+      case 'ArrowDown': {
+        const panFactor = this._settings.interaction.keyboard.panFactor;
+        const dragDownTransform = this._renderer.getPanTransform(PanDirectionType.DOWN, panFactor);
+        this._applyTransformation(dragDownTransform);
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  _zoomOut = (zoomOutFactor: number) => {
+    const transform = this._renderer.getZoomTransform(zoomOutFactor);
+    this._d3Zoom.scaleTo(select(this._canvas), transform.k);
+  };
+
+  _zoomIn = (zoomInFactor: number) => {
+    const transform = this._renderer.getZoomTransform(zoomInFactor);
+    this._d3Zoom.scaleTo(select(this._canvas), transform.k);
+  };
+
+  _applyTransformation = (transform: ZoomTransform, callback?: () => void) => {
+    select(this._canvas)
+      .transition()
+      .duration(this._settings.zoomFitTransitionMs)
```

Comment:
Sorry, I totally missed this one in the last PR check. This `this._settings.zoomFitTransitionMs` is actually from the `recenter` function that does fit to zoom.

In order to make this `applyTransformation` generic enough, I propose having a second argument `options` (that will currently have `transitionMs` (+ callback) only. Something like this:

```typescript
interface IApplyTransitionOptions {
  transitionMs: number;
  callback: () => void;
}

_applyTransformation = (transform: ZoomTransform, options?: Partial<IApplyTransitionOptions>) => {
  const transitionMs = options?.transitionMs ?? DEFAULT_TRANSITION_MS;
  ...
```

Then, the recenter can send its `this._settings.zoomFitTransitionMs` and `callback`. The zoom, and pan can send their `transitionMs`. This `transitionMs` can be even added to the settings with a default value that is lower than 200, maybe 100 or 50. 200 seems a bit slow for expected keyboard movement.

Reply:
I've implemented your suggested approach to enhance the generic nature of the `applyTransformation` function. 

Thank you for your valuable input.